compaction/compaction_manager.cc

```diff
@@ -676,12 +676,16 @@ sstables_task_executor::~sstables_task_executor() {
 }
 
 future<compaction_manager::compaction_stats_opt> compaction_task_executor::run_compaction() noexcept {
-    try {
-        _compaction_done = do_run();
-        return compaction_done();
-    } catch (...) {
-        return current_exception_as_future<compaction_manager::compaction_stats_opt>();
-    }
+    return do_run().then_wrapped([&] (auto f) {
+        try {
+            auto res = f.get();
+            _compaction_done.set_value(res);
```

Comment:
What if someone called `compaction_done()` and is waiting for the returned future to resolve, but `run_compaction()` _isn't called_ for whatever reason? There should be somewhere a call to `_compaction_done.set_exception(compaction_aborted())` or something, but I can't find one